atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/sweep/CassandraGetCandidateCellsForSweepingImpl.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.cassandra.sweep;
+
+import java.util.Iterator;
+import java.util.List;
+
+import org.apache.cassandra.thrift.ConsistencyLevel;
+
+import com.google.common.collect.AbstractIterator;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+import com.palantir.atlasdb.keyvalue.api.CandidateCellForSweeping;
+import com.palantir.atlasdb.keyvalue.api.CandidateCellForSweepingRequest;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.keyvalue.api.ImmutableCandidateCellForSweeping;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.keyvalue.api.Value;
+import com.palantir.atlasdb.keyvalue.cassandra.CassandraKeyValueServices;
+import com.palantir.atlasdb.keyvalue.cassandra.paging.CassandraRawCellValue;
+import com.palantir.atlasdb.keyvalue.cassandra.paging.CellPager;
+import com.palantir.common.annotation.Output;
+import com.palantir.util.Pair;
+
+import gnu.trove.list.TLongList;
+import gnu.trove.list.array.TLongArrayList;
+
+/*
+ * Simply use the CellPager to iterate over raw cells in a table and group the returned entries by the cell key.
+ */
+public class CassandraGetCandidateCellsForSweepingImpl {
+    private final CellPager cellPager;
+
+    public CassandraGetCandidateCellsForSweepingImpl(CellPager cellPager) {
+        this.cellPager = cellPager;
+    }
+
+    public Iterator<List<CandidateCellForSweeping>> getCandidateCellsForSweeping(
+            TableReference tableRef,
+            CandidateCellForSweepingRequest request,
+            ConsistencyLevel consistencyLevel) {
+        Iterator<List<CassandraRawCellValue>> rawIter = cellPager.createCellIterator(
+                tableRef,
+                request.startRowInclusive(),
+                request.batchSizeHint().orElse(100),
```

Comment:
This could be broken out into a named constant

I'm also scared that almost nothing will set this batch hint, though I haven't traced all of the codepaths that create `CandidateCellForSweepRequest`s
The ones I did weren't promising, and I imagine 100 will be pretty safe but unnecessarily quite slow for wide-row dynamic column rows with small cells